scaletest/createworkspaces/run_test.go

```diff
@@ -235,3 +329,40 @@ func Test_Runner(t *testing.T) {
 		require.ErrorContains(t, err, "test error")
 	})
 }
+
+// Since the runner creates the workspace on it's own, we have to keep
+// listing workspaces until we find it, then wait for the build to
+// finish, then start the agents.
+func eventuallyStartFakeAgent(ctx context.Context, t *testing.T, client *codersdk.Client, agentToken string) {
+	t.Helper()
+	var workspace codersdk.Workspace
+	for {
+		res, err := client.Workspaces(ctx, codersdk.WorkspaceFilter{})
+		if !assert.NoError(t, err) {
+			return
+		}
+		workspaces := res.Workspaces
+
+		if len(workspaces) == 1 {
+			workspace = workspaces[0]
+			break
+		}
+
+		time.Sleep(100 * time.Millisecond)
+	}
+
+	coderdtest.AwaitWorkspaceBuildJob(t, client, workspace.LatestBuild.ID)
+
+	agentClient := agentsdk.New(client.URL)
+	agentClient.SetSessionToken(agentToken)
+	agentCloser := agent.New(agent.Options{
+		Client: agentClient,
+		Logger: slogtest.Make(t, &slogtest.Options{IgnoreErrors: true}).
+			Named("agent").Leveled(slog.LevelWarn),
+	})
+	t.Cleanup(func() {
```

Comment:
Calling `t.Cleanup` asynchronously (potentially after test ended, and other times the order of cleanups changes) seems like it could result in a bad time. How about this kind of usage:

```go
cleanup := goEventuallyStartFakeAgent(ctx, t, ...)
t.Cleanup(cleanup)
```

This forces us to write the function in a way that avoids the aforementioned issue.

(The cleanup function should not return before all resources are cleaned up.)